internal/generator/ecdsa/template/ecdsa.go.tmpl

```diff
@@ -0,0 +1,219 @@
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/sha512"
+	"io"
+	"math/big"
+
+	"github.com/consensys/gnark-crypto/ecc/{{ .Name }}"
+	"github.com/consensys/gnark-crypto/ecc/{{ .Name }}/fr"
+)
+
+// PublicKey represents an ECDSA public key
+type PublicKey struct {
+	Q {{ .CurvePackage }}.G1Affine
+}
+
+// PrivateKey represents an ECDSA private key
+type PrivateKey struct {
+	PublicKey
+	Secret big.Int
+}
+
+// Signature represents an ECDSA signature
+type Signature struct {
+	R, S big.Int
```

Comment:
Usually the convention is to use *big.Int. There are slices in the struct anyway, so is always heap-allocated.